src/IO/ADIOS/ParallelADIOS1IOHandler.cpp

```diff
@@ -314,6 +316,11 @@ ParallelADIOS1IOHandlerImpl::open_read(std::string const& name)
 
     return f;
 }
+
+#define ADIOS1IOHandlerImpl ParallelADIOS1IOHandlerImpl
+#include "ADIOS1IOCommon.cpp"
+#undef ADIOS1IOHandlerImpl
```

Comment:
When you *think* you've seen all the ways of sharing code between classes...

Reply:
yeah, that's my secret present for you.
did I mention it's the *worst hack* ever?

Reply:
Note that inheritance does not work since *identical symbols do not mean identical implementation* thanks to the mock...

Reply:
I could inherit from yet another baseclass though, that we then hide again in symbols and compile twice... ahhhh... it must stop.

Reply:
The reason for the duplicated code is not too diffcult to understand. Just maybe replace ```ADIOS1IOHandlerImpl``` with ```CommonImpl``` in ```"ADIOS1IOCommon.cpp"``` and then do the ```#define```s in both handlers?

Reply:
Yeah, thought so, too!